src/sidebar/util/groups.js

```diff
@@ -20,7 +23,48 @@ function combineGroups(userGroups, featuredGroups) {
   featuredGroups.forEach(group => (group.isMember = false));
   userGroups.forEach(group => (group.isMember = true));
 
-  return userGroups.concat(featuredGroups);
+  const groups = userGroups.concat(featuredGroups);
+
+  // Add isSelectable property indicating whether a group can be
+  // annotated in at this particular url.
+  groups.forEach(group => (group.isSelectable = isSelectable(group, uri)));
+
+  // Remove unneeded scopes data from group to save memory.
+  groups.forEach(group => delete group.scopes);
+
+  return groups;
+}
+
+function isSelectable(group, uri) {
+  // If the group is scoped, the scoping is enforced,
+  // and the uri patterns don't inlude this page's uri
+  // the group is not selectable, otherwise it is.
+  if (
+    group.scoped &&
+    group.scopes.enforced &&
```

Comment:
What will happen if this function is accidentally called with a group for which scope information has not been fetched?

Reply:
Good catch! I intended to cover that case in the tests but..looks like I forgot the there-is-no-scopes-at-all case so I've added a test and logic for that.